### PR TITLE
Update from commons-configuration to commons-configuration2

### PR DESCRIPTION
**Pull request by henning-gerhardt**
This PR includes:
- Updating / Upgrading from commons-configuration to commons-configuration2
-  Adding new test class for OPACConfig class
-  Replacing used exception class from dependency on throwing it with a exception from Kitodo namespace

### CODE REVIEW

File: Kitodo-API/src/test/java/org/kitodo/config/OPACConfigTest.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    @Test
+    void isPrestructuredImportIsReturingCorrectValue() {
+        assertFalse(OPACConfig.isPrestructuredImport("K10Plus"));
+    }
+
+    @Test
+    void getXsltMappingFilesIsThrowingExceptionOnMissingMappingFilesEntry() {
+        Exception exception = assertThrows(ConfigurationRuntimeException.class,
+                () -> { List<String> result = OPACConfig.getXsltMappingFiles("K10Plus");
```
[the remark this answers is not in the document]
Comment (henning-gerhardt) [reply Kitodo-API/src/test/java/org/kitodo/config/OPACConfigTest.java:74]: Is not ignoring / not assigning a return value from a method a better solution instead of not using it? Even inside a test where the check is that a exception is thrown and the method return value is not relating and even not available?

[the remark this answers is not in the document]
Comment (henning-gerhardt) [reply Kitodo-API/src/test/java/org/kitodo/config/OPACConfigTest.java]: Fixing it while rebasing.

File: Kitodo-API/src/main/java/org/kitodo/config/Config.java [hunk 2]
```diff
@@ -41,20 +42,26 @@ static PropertiesConfiguration getConfig(String configFile) {
...
+                        ReloadingFileBasedConfigurationBuilder<PropertiesConfiguration> builder =
+                            new ReloadingFileBasedConfigurationBuilder<>(PropertiesConfiguration.class)
+                                .configure(new Parameters().properties()
+                                    .setFileName(configFile)
+                                    .setThrowExceptionOnMissing(true)
+                                    .setListDelimiterHandler(new DefaultListDelimiterHandler('&'))
+                                    .setIncludesAllowed(false));
+                        // Register an event listener for triggering reloading checks
+                        builder.addEventListener(ConfigurationBuilderEvent.CONFIGURATION_REQUEST,
+                            (EventListener) event -> builder.getReloadingController().checkForReloading(null));
```
Comment (solth) [inline Kitodo-API/src/main/java/org/kitodo/config/Config.java:56]:
```suggestion
                            event -> builder.getReloadingController().checkForReloading(null));
```
Is this cast necessary? I first extended it to `(EventListener<ConfigurationBuilderEvent>) event` because the IDE marked it as an "Unchecked assignment" but then it would tell me that the cast is redundant altogether. I removed the typecast and it still works.

File: Kitodo-API/src/main/java/org/kitodo/config/OPACConfig.java [hunk 3]
```diff
@@ -518,6 +516,15 @@ public static XMLConfiguration getKitodoOpacConfiguration() throws Configuration
...
-        return new XMLConfiguration(kitodoConfigOpacFile.getFile());
+        // Create and initialize the builder
+        ReloadingFileBasedConfigurationBuilder<XMLConfiguration> builder =
+            new ReloadingFileBasedConfigurationBuilder<>(XMLConfiguration.class)
+                .configure(new Parameters().xml()
+                    .setFile(kitodoConfigOpacFile.getFile())
+                    .setListDelimiterHandler(new DefaultListDelimiterHandler('&')));
+        // Register an event listener for triggering reloading checks
+        builder.addEventListener(ConfigurationBuilderEvent.CONFIGURATION_REQUEST,
+            (EventListener) event -> builder.getReloadingController().checkForReloading(null));
```
Comment (solth) [inline Kitodo-API/src/main/java/org/kitodo/config/OPACConfig.java:527]:
```suggestion
            event -> builder.getReloadingController().checkForReloading(null));
```
Is this cast necessary? See above.

File: Kitodo-API/src/test/java/org/kitodo/config/OPACConfigTest.java
Comment (solth) [inline Kitodo-API/src/test/java/org/kitodo/config/OPACConfigTest.java:1]: Thanks for adding this new test file!

File: Kitodo-DataManagement/src/main/java/org/kitodo/config/ConfigMain.java [hunk 4]
```diff
@@ -42,16 +46,23 @@ private static PropertiesConfiguration getConfig() {
...
+                        // Create and initialize the builder
+                        ReloadingFileBasedConfigurationBuilder<PropertiesConfiguration> builder =
+                            new ReloadingFileBasedConfigurationBuilder<>(PropertiesConfiguration.class)
+                                .configure(new Parameters().properties()
+                                    .setFileName(CONFIG_FILE)
+                                    .setListDelimiterHandler(new DefaultListDelimiterHandler('&'))
+                                    .setIncludesAllowed(false));
+                        // Register an event listener for triggering reloading checks
+                        builder.addEventListener(ConfigurationBuilderEvent.CONFIGURATION_REQUEST,
+                            (EventListener) event -> builder.getReloadingController().checkForReloading(null));
```
Comment (solth) [inline Kitodo-DataManagement/src/main/java/org/kitodo/config/ConfigMain.java:59]:
```suggestion
                            event -> builder.getReloadingController().checkForReloading(null));
```
Is this cast necessary? See above.

File: Kitodo-Docket/src/main/java/org/kitodo/docket/ExportXmlLog.java [hunk 5]
```diff
@@ -382,9 +386,16 @@ private HashMap<String, String> getXmlPathFromConfig(String xmlPath) {
...
-                config.setReloadingStrategy(new FileChangedReloadingStrategy());
+                // Create and initialize the builder
+                ReloadingFileBasedConfigurationBuilder<XMLConfiguration> builder =
+                    new ReloadingFileBasedConfigurationBuilder<>(XMLConfiguration.class)
+                        .configure(new Parameters().xml()
+                            .setFile(file)
+                            .setListDelimiterHandler(new DefaultListDelimiterHandler('&')));
+                // Register an event listener for triggering reloading checks
+                builder.addEventListener(ConfigurationBuilderEvent.CONFIGURATION_REQUEST,
+                    (EventListener) event -> builder.getReloadingController().checkForReloading(null));
```
Comment (solth) [inline Kitodo-Docket/src/main/java/org/kitodo/docket/ExportXmlLog.java:397]:
```suggestion
                    event -> builder.getReloadingController().checkForReloading(null));
```
See above.

File: Kitodo-ImageManagement/src/test/resources/kitodo_config.properties
Comment (solth) [inline Kitodo-ImageManagement/src/test/resources/kitodo_config.properties:1]: Why is this file necessary when it only contains comments?

File: Kitodo/src/main/java/de/unigoettingen/sub/search/opac/ConfigOpac.java [hunk 6]
```diff
@@ -43,13 +47,20 @@ private static XMLConfiguration getConfig() throws FileNotFoundException {
...
-            config = new XMLConfiguration(opacConfiguration.getFile());
+            // Create and initialize the builder
+            ReloadingFileBasedConfigurationBuilder<XMLConfiguration> builder =
+                new ReloadingFileBasedConfigurationBuilder<>(XMLConfiguration.class)
+                    .configure(new Parameters().xml()
+                        .setFile(opacConfiguration.getFile())
+                        .setListDelimiterHandler(new DefaultListDelimiterHandler('&')));
+            // Register an event listener for triggering reloading checks
+            builder.addEventListener(ConfigurationBuilderEvent.CONFIGURATION_REQUEST,
+                (EventListener) event -> builder.getReloadingController().checkForReloading(null));
```
Comment (solth) [inline Kitodo/src/main/java/de/unigoettingen/sub/search/opac/ConfigOpac.java:58]:
```suggestion
                event -> builder.getReloadingController().checkForReloading(null));
```

File: Kitodo/src/main/java/org/kitodo/config/ConfigProject.java [hunk 7]
```diff
@@ -61,13 +65,20 @@ public ConfigProject(String projectTitle) throws IOException {
...
-            this.config = new XMLConfiguration(configFile.getAbsolutePath());
+            // Create and initialize the builder
+            ReloadingFileBasedConfigurationBuilder<XMLConfiguration> builder =
+                new ReloadingFileBasedConfigurationBuilder<>(XMLConfiguration.class)
+                    .configure(new Parameters().xml()
+                        .setFile(configFile.getFile())
+                        .setListDelimiterHandler(new DefaultListDelimiterHandler('&')));
+            // Register an event listener for triggering reloading checks
+            builder.addEventListener(ConfigurationBuilderEvent.CONFIGURATION_REQUEST,
+                (EventListener) event -> builder.getReloadingController().checkForReloading(null));
```
Comment (solth) [inline Kitodo/src/main/java/org/kitodo/config/ConfigProject.java:76]:
```suggestion
                event -> builder.getReloadingController().checkForReloading(null));
```

File: pom.xml [hunk 8]
```diff
@@ -46,9 +46,7 @@
         <awaitility.version>4.0.1</awaitility.version>
         <com.h2database.h2.version>2.2.224</com.h2database.h2.version>
         <commons-codec.version>1.15</commons-codec.version>
-        <commons-configuration.version>1.10</commons-configuration.version>
         <commons-io.version>2.11.0</commons-io.version>
-        <commons-lang3.version>3.12.0</commons-lang3.version>
```
Comment (solth) [inline pom.xml:51]: Why did you remove the variable instead of updating the version here? I think it is clearer to have all version numbers in one place instead of gathering from all over the code. Wouldn't you agree?

File: Kitodo-ImageManagement/src/test/resources/kitodo_config.properties
Comment (henning-gerhardt) [reply Kitodo-ImageManagement/src/test/resources/kitodo_config.properties:1]: I got errors as this file was not existing while executing the tests. The code is awaiting a configuration file and is complaining about if the file is missing. But I will look into this if this is still the case.

File: pom.xml (hunk 8 above)
Comment (henning-gerhardt) [reply pom.xml:51]: This (and a lot of other) variable is not needed anymore as the usage is on one place. In the past the used versions was moved into the main pom.xml file and the module pom.xml files rely on the version information of the main pom.file. I can re-add this variable if you prefer this.

File: Kitodo-API/src/main/java/org/kitodo/config/Config.java (hunk 2 above)
Comment (henning-gerhardt) [reply Kitodo-API/src/main/java/org/kitodo/config/Config.java:56]: The `EventListener` casting is not needed. Thanks for hinting this.

File: Kitodo-ImageManagement/src/test/resources/kitodo_config.properties
Comment (henning-gerhardt) [reply Kitodo-ImageManagement/src/test/resources/kitodo_config.properties:1]: File is still needed. I can check in an empty file (only license header but no real content) but this did not look right. Whats you opinion @solth ?

Comment (solth) [reply Kitodo-ImageManagement/src/test/resources/kitodo_config.properties:1]: I am fine with this file being included the way it is. No change required. I was just curious if it was added accidentally.

File: pom.xml (hunk 8 above)
Comment (henning-gerhardt) [reply pom.xml:51]: I re-added the variable. Cleanup such one-use variable should be done in future.

**Review by solth — APPROVED** [review]
